src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -21,6 +21,13 @@ public override BoundNode VisitConversion(BoundConversion node)
                 return RewriteInterpolatedStringConversion(node);
             }
 
+            if (node.ConversionKind is ConversionKind.SwitchExpression or ConversionKind.ConditionalExpression)
```

Comment:
Was this change necessary? For target-typed `new()`, I think the expression is also wrapped in a `BoundConversion` during initial binding, but we don't erase that conversion during lowering.

Reply:
Yes, it's necessary. Emit asserts if this isn't removed.

---
In reply to: [595617051](https://github.com/dotnet/roslyn/pull/51791#discussion_r595617051) [](ancestors = 595617051)